common/achievements.h

```diff
@@ -42,58 +42,145 @@ enum AchievementsPlatform {
 
 
 /**
- * Per-game achievements information structure item.
+ * Information structure for game-specific achievements.
  */
 struct AchievementDescription {
-	const char *id;            // achievement internal id, e.g. "ACHIEVEMENT_TIMING"
-	bool isHidden;             // achievement is hidden
-	const char *title;         // achievement displayed text, e.g. "Marathon Runner"
-	const char *comment;       // optional achievement hint / comment, e.g. "Finish the game in less than 4 hours"
+	const char *id;            //!< Achievement internal ID, such as "ACHIEVEMENT_TIMING".
+	bool isHidden;             //!< Whether the achievement is hidden.
+	const char *title;         //!< Achievement displayed text, such as "Marathon Runner".
+	const char *comment;       //!< Optional achievement hint or comment, such as "Finish the game in less than 4 hours".
 };
 
 
 /**
- * Per-game achievements information structure item.
+ * Information structure for platform-specific achievements.
  */
 struct AchievementsInfo {
-	Common::AchievementsPlatform platform;              // achievements platform, e.g. STEAM_ACHIEVEMENTS
-	Common::String appId;                               // achievements application ID of given platform
-	Common::Array<AchievementDescription> descriptions; // descriptions of all game achievements
+	Common::AchievementsPlatform platform;              //!< Achievements platform, such as "STEAM_ACHIEVEMENTS".
+	Common::String appId;                               //!< Achievements application ID of the given platform.
+	Common::Array<AchievementDescription> descriptions; //!< Descriptions of all game achievements.
 
 	AchievementsInfo() {platform = Common::UNK_ACHIEVEMENTS;}
 };
 
+/**
+ * Class for manipulating the achievements.
+ *
+ * Use the Achievements Manager class to edit the in-game achievements.
+ */
 
 class AchievementsManager : public Singleton<AchievementsManager> {
 public:
 	AchievementsManager();
 	~AchievementsManager();
-
+    
+	/**
+     * Set a platform and application ID as active domain.
+	 *
+	 * @param[in] platform Achievements platform.
```

Comment:
The big question is whether we should include documentation for all parameters in every method. I think this is a good practice but it definitely adds a lot of doxygen noise to the header files. If parameters were not documented, most of these class members could be documented with ``//!<`` in the same line, instead of adding a preceding doxygen block.